doc/man3/SSL_CTX_set1_curves.pod

```diff
@@ -75,17 +75,76 @@ to downgrade connections to a weaker algorithm.
 =back
 
 SSL_CTX_set1_groups_list() sets the supported groups for B<ctx> to
-string B<list>. The string is a colon separated list of group names, for example
-"P-521:P-384:P-256:X25519:ffdhe2048". The groups are used as in
+string B<list>. As an extension to SSL_CTX_set1_groups(), the names of the
+groups rather than their NIDs are used. The groups are used as in
 SSL_CTX_set1_groups(), described above. Currently supported groups for
 B<TLSv1.3> are B<P-256>, B<P-384>, B<P-521>, B<X25519>, B<X448>,
 B<brainpoolP256r1tls13>, B<brainpoolP384r1tls13>, B<brainpoolP512r1tls13>,
 B<ffdhe2048>, B<ffdhe3072>, B<ffdhe4096>, B<ffdhe6144> and B<ffdhe8192>. Support
-for other groups may be added by external providers, however note the discussion
-on TLS 1.3 selection criteria above. If a group name is preceded with the C<?>
-character, it will be ignored if an implementation is missing. The "DEFAULT"
-group name is used to select the default selection of groups. Group names that
-are preceded with the C<-> character will be removed from the selected groups.
+for other groups may be added by external providers.
+The group can be either the B<NIST> name (e.g. B<P-256>), some other commonly
+used name where applicable (e.g. B<X25519>, B<ffdhe2048>) or an OpenSSL OID name
+(e.g. B<prime256v1>). Group names are case sensitive.
+
+The B<list> can be used to define several group tuples of comparable security
+levels, influence the number of key shares sent by a client, ignore unknown
+groups, add the OpenSSL default set of groups or allows to remove a group from
+the list.
+
+In it's simplest form, the string B<list> is just a colon separated list
+of group names, for example "P-521:P-384:P-256:X25519:ffdhe2048". The first
+group listed will also be used for the B<key_share> sent by a client in a
+TLSv1.3 B<ClientHello>. For servers note the discussion above. The list should
+be in order of preference with the most preferred group first.
+
+Group tuples of comparable security are defined by wrapping such groups with B<{..}>
+brackets. Key shares to be sent by a client are defined by preceding the group
+name with C<*>, for example
+{P-521:*P-256}:{*P-384}:{*X25519:P-384:ffdhe2048}.
+General syntax for the string B<list> is:
+C<[{] [-][?][*]group1 rep[:[-][?][*]groupN] [}] rep[: {[-][?][*]groupM rep[:[-][?][*]groupP]} ]>
```

Comment:
```suggestion
Group tuples of comparable security are defined by enclosing each tuple in B<{..}> braces.
Keyshares to be sent by a client are specified by prepending a C<*> to the group name,
for example:
{P-521:*P-256}:{*P-384}:{*X25519:P-384:ffdhe2048}.
The general syntax for the string B<list> is:
C<[{] [-][?][*]group1 rep[:[-][?][*]groupN] [}] rep[: {[-][?][*]groupM rep[:[-][?][*]groupP]} ]>
```
I've not seen much use of "rep:[...]" in specifying ABNF grammars. Much more common
are `*(...)`, `1*(...)`, ...
It may be more appropriate to use a more widely used grammar?

Reply:
@vdukhovni I think the idea here was to have a simple compact way to describe the structure of the list string. rep[] was chosen because asterisk has specific meaning within the string structure, so it might cause confusion to just replace rep[] with *(). Happy to discuss more options if you feel that a more formal notation or alternative would be better and easier for end users